### PR TITLE
use requirements.txt file for dependencies

### PR DESCRIPTION
**Pull request by mzfr**
We are planning to use pipenv for maintaining dependencies but until then we'll be using requirements.txt file.
.travis.yml is updated for installing all the requirements from requirements.txt file

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pytest
```
Comment (Razzeee) [inline requirements.txt]: Please remove pytest and pycodestyle as we only need them in the travis.yml

Comment (mzfr) [reply requirements.txt]: about pytest and pycodestyle: we'll have to keep the pycodestyle in the file else we'll have error about "pycodestyle not installed" or something like that ?

Comment (mzfr) [reply requirements.txt]: Done

File: .travis.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ matrix:
 
 # command to install dependencies
 install:
-  - pip install pytest Pillow pycodestyle
+  - pip install -r requirements.txt
```
Comment (Razzeee) [inline .travis.yml]: these are missing pytest and pycodestyle now

Comment (Razzeee) [reply .travis.yml]: Hrm, seems to be included in the python image then. The docs even reference pytest https://docs.travis-ci.com/user/languages/python/
So we should be good

**Pull request merged**